### PR TITLE
Fix DateTime serialization.

### PR DESCRIPTION
**Pull request by michaelcfanning**
@lgolding @rtaket

### CODE REVIEW

File: src/Sarif.Driver/Sdk/SarifLogger.cs [hunk 1]
```diff
@@ -90,15 +91,8 @@ public SarifLogger(string outputFilePath, bool verbose, ToolInfo toolInfo, RunIn
 
         }
 
-        public SarifLogger(TextWriter writer, bool verbose, ToolInfo toolInfo, RunInfo runInfo)
+        public SarifLogger(TextWriter textWriter, bool verbose, ToolInfo toolInfo, RunInfo runInfo) : this(textWriter, verbose)
```
Comment (michaelcfanning) [inline src/Sarif.Driver/Sdk/SarifLogger.cs]: @lgolding this useful change you suggested in a previous PR was zapped in an unfortunate merge incident that is still too painful for me to discuss in much detail.

File: src/Sarif/Readers/DateTimeConverter.cs [hunk 2]
```diff
@@ -27,13 +27,15 @@ public override bool CanConvert(Type objectType)
...
+            if (reader.Value is DateTime) { return reader.Value; }
+
             return DateTime.Parse((string)reader.Value, CultureInfo.InvariantCulture);
         }
 
         public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
         {
             string formattedDate = ((DateTime)value).ToString(DateTimeFormat, CultureInfo.InvariantCulture);
-            writer.WriteRawValue(@"""" + value + @"""");
+            writer.WriteRawValue(@"""" + formattedDate + @"""");
```
Comment (michaelcfanning) [inline src/Sarif/Readers/DateTimeConverter.cs]: oops.

File: src/Sarif/Readers/DateTimeConverter.cs [hunk 3]
```diff
@@ -27,13 +27,15 @@ public override bool CanConvert(Type objectType)
 
         public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
         {
+            if (reader.Value is DateTime) { return reader.Value; }
```
Comment (michaelcfanning) [inline src/Sarif/Readers/DateTimeConverter.cs]: JSON.NET appears to have some built-in DateTime detection so, at least in the current test cases, the object is already converted. We still need the converter, to output the correct format. But I haven't yet confirmed whether this conditional is always satisfied (in which case we can simplify the implementation to simply return reader.Value. Will follow-up

File: src/Sarif.Driver/Sdk/SarifLogger.cs (hunk 1 above)
Comment (lgolding) [inline src/Sarif.Driver/Sdk/SarifLogger.cs]: :)

**Comment by lgolding** [comment]
:+1:

**Pull request merged**